lib/dry/monitor.rb

```diff
@@ -1,20 +1,38 @@
 # frozen_string_literal: true
 
-require "dry/monitor/logger"
-require "dry/monitor/notifications"
 require "dry/core/extensions"
+require "dry/core/constants"
+require "zeitwerk"
+
 
 module Dry
   module Monitor
     extend Dry::Core::Extensions
+    include Dry::Core::Constants
 
     register_extension(:rack) do
       require "rack/utils"
-      require "dry/monitor/rack/logger"
     end
 
     register_extension(:sql) do
-      require "dry/monitor/sql/logger"
```

Comment:
Since autoloading makes these files load when referenced, how should we handle the extension APIs for this gem? Maybe log a deprecation warning?

Reply:
Maybe reference the constant here?